compiler+runtime/src/cpp/jank/read/lex.cpp

```diff
@@ -289,7 +336,15 @@ namespace jank::read
       }
 
       auto const token_start(pos);
-      switch(file[token_start])
+      auto const oc = convert_to_codepoint(file.substr(token_start), token_start);
+      char32_t start{};
+      size_t len{};
```

Comment:
We have a struct for this! You end up doing `c` and `len` (in some form) **six** different times in this file. Just use our `codepoint` struct instead.